app/viewmodels/checkAnswers/TraderReferenceSummary.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels.checkAnswers
+
+import controllers.routes
+import models.{CheckMode, UserAnswers}
+import pages.TraderReferencePage
+import play.api.i18n.Messages
+import play.twirl.api.HtmlFormat
+import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist.SummaryListRow
+import viewmodels.govuk.summarylist._
+import viewmodels.implicits._
+
+object TraderReferenceSummary {
+
+  def row(answers: UserAnswers)(implicit messages: Messages): Option[SummaryListRow] =
```

Comment:
Confused - what is TraderReferenceSummary for?

Reply:
@liandra-giacomine : All these auto generated summary objects are used in cya page-load to automatically display the data without manual effort. This is generated for all pages with check modes. Pasting the code example from other cya controller

 def onPageLoad(): Action[AnyContent] = (identify andThen getData andThen requireData) { implicit request =>
    TraderProfile.build(request.userAnswers, request.eori) match {
      case Right(_)     =>
        val list = SummaryListViewModel(
          rows = Seq(
            UkimsNumberSummary.row(request.userAnswers),
            HasNirmsSummary.row(request.userAnswers),
            NirmsNumberSummary.row(request.userAnswers),
            HasNiphlSummary.row(request.userAnswers),